### PR TITLE
BUG: Series[Interval].fillna(mismatched) coerce instead of raise

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Achieves consistency across methods in coercion behavior for IntervalDtype.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@jreback most recent commit fixes a mypy complaint on main

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -282,6 +282,7 @@ Indexing
 Missing
 ^^^^^^^
 - Bug in :meth:`Series.fillna` and :meth:`DataFrame.fillna` with ``downcast`` keyword not being respected in some cases where there are no NA values present (:issue:`45423`)
+- Bug in :meth:`Series.fillna` and :meth:`DataFrame.fillna` with :class:`IntervalDtype` and incompatible value raising instead of casting to a common (usually object) dtype (:issue:`??`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.5.0.rst:285]: `??` -> `45796` (follow up reminder)

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
got it minor comment u can do in a follow up?

**Comment by jbrockmendel** [comment]
> got it minor comment u can do in a follow up?

yes

File: doc/source/whatsnew/v1.5.0.rst (hunk 1 above)
Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.5.0.rst:285]: included in #45940

**Pull request merged**